### ISSUE TITLE
avay phone can't use phone button

### ISSUE THREAD

**Issue by nikohpng** [willamowius/gnugk#14]
## Description
I have implemented the registration of the avaya phone, but currently the phone cannot use the number key, and the function key on the left is responding.

I have do it [avaya-support](https://fossies.org/linux/gnugk/docs/avaya/avaya-support.txt). If there's anything missing from me, please  contact me.

## Config
gnugk
```ini
; this is a basic startup configuration for GnuGk with Avaya support
[Gatekeeper::Main]
Name=DEFINITY-GK
TimeToLive=600
CompareAliasType=0
CompareAliasCase=0
EndpointIDSuffix=
Home=192.168.1.101

; restrict access to the status port by an IP address
[GkStatus::Auth]
rule=explicit
; allow local access
127.0.0.1=allow
default=forbid
Shutdown=allow

[RoutedMode]
GKRouted=1
H245Routed=1
CallSignalPort=1720
AcceptUnregisteredCalls=1
DropCallsByReleaseComplete=1
SupportNATedEndpoints=1
EnableH46018=1
PregrantARQ=1
GenerateCallProceeding=1
H245TunnelingTranslation=1
Q931PortRange=20000-29999

[Proxy]
Enable=1
ProxyAlways=0
; GWID must unique identify Asterisk OOH323 registration (ACE is a default)
; see Asterisk "ooh323.conf" for example
AsteriskGWID=ACE
; GWDN must unique identify automatically dialed digits into context with
; DISA application, for providing emulation of Avaya switch functionality
; see Asterisk "extensions.ael" for example
AsteriskGWDN=1999

[RoutingPolicy]
default=explicit,internal,srv,dns

[LogFile]
Rotate=Daily
RotateTime=23:59

[Gatekeeper::Auth]
SimplePasswordAuth=alternative;RRQ
FileIPAuth=required;RRQ
default=allow

[SimplePasswordAuth]
KeyFilled=123
PasswordTimeout=0
; this entry will be automatically added by addpwd application from GnuGk
; it enciphers password for subsciber with extension 1000 (auth UserID)
1000=UbvtLZfPtQw=

[FileIPAuth]
; permit anything from local host
192.168.1.101/32=allow

[RasSrv::GWPrefixes]
; route all traffic for prefix 19 to our Asterisk GWID (ACE)
ACE=19:=1

[RasSrv::RRQFeatures]
OverwriteEPOnSameAddress=allow

[H235]
UseEndpointIdentifier=0
RequireGeneralID=0
FullQ931Checking=0
CheckSendersID=0
VerifyRandomNumber=0
TimestampGracePeriod=7210

```

ooh323.conf
```conf
[general]
;Define the asterisk server h323 endpoint

; Using port 1820 because 1720 already bind in GnuGk
port=1820

; Living on the same address with GnuGk
bindaddr=192.168.1.101

gateway=yes

h245tunneling=no

mediawaitforconnect=yes

; Our GWID and prefix for DISA application
h323id=ACE
e164=1999

callerid=Asterisk

aniasdni=yes

gatekeeper = 192.168.1.101

;logfile=/var/log/asterisk/h323_log

context=default

rtptimeout=12           ; Terminate call if 60 seconds of no RTP activity
                        ; when we're not on hold

;Default - none (lowdelay, thoughput, reliability, mincost, none)
;tos=lowdelay

;amaflags = default
;accountcode=h3230101

disallow=all     ;Note order of disallow/allow is important.
allow=ulaw

dtmfmode=q931keypad

faxdetect = no

directmedia=no

directrtpsetup=no

; set this to your variant
tracelevel=6

```

## Log
```

2023/09/21 17:42:05.524 5       ProxyChannel.cxx(1641)  Q931s   Reading from 192.168.1.33:6362
2023/09/21 17:42:05.524 3       ProxyChannel.cxx(2295)  Q931s   Received: Information CRV=1 from 192.168.1.33:6362
2023/09/21 17:42:05.524 4       ProxyChannel.cxx(2204)  Q931    Received: {
  q931pdu = {
    protocolDiscriminator = 8
    callReference = 1
    from = originator
    messageType = Information
    IE: User-User = {
      28 10 01 00 10 a0 01 00  18 01 00 0a 60 86 48 01   (...........`.H.
      86 f8 72 04 02 0a 0a 0a  38 47 00 c0 0b 36 60 08   ..r.....8G...6`.
      40 00                                              @.
    }
  }
  h225pdu = {
    h323_uu_pdu = {
      h323_message_body = empty <<null>>
      h245Tunneling = false
      nonStandardControl = 1 entries {
        [0]={
          nonStandardIdentifier = object 2.16.840.1.113778.4.2.10
          data =  10 octets {
            0a 38 47 00 c0 0b 36 60  08 40                     .8G...6`.@
          }
        }
      }
    }
  }
}
2023/09/21 17:42:05.524 3             gkauth.cxx(1021)  GKAUTH  default Q931 check ok
2023/09/21 17:42:05.524 3       ProxyChannel.cxx(7603)  Q931s   Avaya: INFO+CCMS
2023/09/21 17:42:05.524 3       ProxyChannel.cxx(8391)  Q931s   Avaya: CCMS EP TERM ID
192.168.1.33:6362|1000:dialedDigits|terminal|3117827206|Onhook
Thu, 21 Sep 2023 17:42:04 +08:00 C(1/1/1)  <3> bw:0/-1 vendor:IP_Phone 1.3120 specific:Onhook CCMS:
Call No. 0 | CallID 00 00 00 00 00 00 10 00 00 00 00 00 c0 a8 01 21 | 0 | 0
Dial invalid address
ACF|192.168.1.33:6362|3117827206|1|invalid address||false|00-00-00-00-00-00-10-00-00-00-00-00-c0-a8-01-21|Direct;
ACF|0.0.0.0:0| |32769|invalid address||true|00-00-00-00-00-00-10-00-00-00-00-00-c0-a8-01-21|Direct;
```

## Env
- [x] gnugk-5_9
- [x] asterisk-16.5.0
- [x] h323plus-1_27_2
- [x] ptlib-2_10_9_4
- [x] avaya1608i  1.3.12

## Last
I have poor understand of h323. Anyone can give me some advice?

**Comment by nikohpng** [comment]
[avaya1608-t.pcap](https://drive.google.com/file/d/1G9VNxeGV6jOhPHUyr2_htt45m17QL2Qq/view?usp=sharing): this regist to gnugk
[1608avaya](https://drive.google.com/file/d/1W0UnEjvLEXCYxyop9DNhgR06mNRFKfZw/view?usp=sharing): this regist to avaya office
[debug.log](https://drive.google.com/file/d/1GjF6nBnxV7kGut509JJt8aVDKuoLkNak/view?usp=sharing)

**Comment by kprokazov** [comment]
> [avaya1608-t.pcap](https://drive.google.com/file/d/1G9VNxeGV6jOhPHUyr2_htt45m17QL2Qq/view?usp=sharing): this regist to gnugk [1608avaya](https://drive.google.com/file/d/1W0UnEjvLEXCYxyop9DNhgR06mNRFKfZw/view?usp=sharing): this regist to avaya office [debug.log](https://drive.google.com/file/d/1GjF6nBnxV7kGut509JJt8aVDKuoLkNak/view?usp=sharing)

Maybe you misunderstood me. I've asked you about:
- 46xxsettings.txt file
- asterisk dialplan
in my previous reply. I don't required now in pcap or debug logs. 16xx/46xx/96xx Avaya series have no call control engine inside phone itself, ALL phone features are managed by PBX CPU via CCMS protocol (H.323 here only a simple transport). Please read carefully https://fossies.org/linux/gnugk/docs/avaya/avaya-support.txt point 5 and higher.

**Comment by nikohpng** [comment]
Sorry, I'm confused.
[46xxsettings.txt](https://drive.google.com/file/d/1-zXpiWXTmMRCnSKWM6Q1SwDumC4vzXxs/view?usp=sharing)
[extensions.ael](https://drive.google.com/file/d/1uAVe8DWDmVN3KacpWqdlCXphC2USpngf/view?usp=sharing)
I am using recommended configurations

**Comment by nikohpng** [comment]
Sorry, this title Possible unclear.
the screen not show any number when i push number button in my 1608i phone.
Of cause, I have recieved like dtmf information in gnugk.
is this normal?

**Comment by kprokazov** [comment]
Seems like some kind of feature regression, can you contact me directly please at kprokazov@gmail.com with detailed logs (-ttttt). I'll compile latest version at my own and check also.

**Comment by nikohpng** [comment]
> Seems like some kind of feature regression, can you contact me directly please at [kprokazov@gmail.com](mailto:kprokazov@gmail.com) with detailed logs (-ttttt). I'll compile latest version at my own and check also.

thank you reply. I have solved. 
We must off-hook to push number button, otherwise we can't see any number in screen

**Comment by willamowius** [comment]
Since you went through the whole installation process now, maybe you can add to the documentation so it is easier for the next person.

Thanks!

**Comment by kprokazov** [comment]
this behaviour is already documented, that there are no AUTO-START feature for outgoing calls.
You MUST lift handset OR press Call-appearance button (#1) or Speaker button.
Only after this session will be properly initiated and you'll listen to dialtone.
BR, K.
